database/migrations/2023_11_12_183753_meeting_data.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+
+    private string $meetingCategoryDictionaryId = 'ce12aa23-a5db-49f3-987b-d2db3ab24a3b';
+    private string $meetingTypeDictionaryId = '4cc5d6b5-5913-4e07-9fcb-09b6f3d3d2a0';
+    private string $meetingStatusDictionaryId = '3865a3c3-0038-4668-9d55-5d05b79d7fcd';
+    private string $meetingResourceDictionaryId = 'fdb0f818-aa1e-4ed9-97cc-2a3cb1b702df';
+    private string $meeting_category_attribute_id = '8111626d-130c-454d-b0c0-9fda9ab9917a';
+    private string $meeting_type_attribute_id = '5f7d5e66-03f9-4bcd-a726-fde82cf98d6f';
+    private string $meeting_status_attribute_id = 'e443e2c2-82fc-41d3-8fda-fe374e5329d3';
+    private string $meeting_resource_attribute_id = 'ff2fb31e-c992-4e2d-b543-f2768ee1c897';
+
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::create('meetings', function (Blueprint $table) {
+            $table->char('id', 36)->collation('ascii_bin')->primary();
+            $table->char('facility_id', 36)->collation('ascii_bin');
+            $table->char('category_dict_id', 36)->collation('ascii_bin');
+            $table->char('type_dict_id', 36)->collation('ascii_bin');
+            $table->string('name')->nullable();
+            $table->text('notes')->nullable();
+            $table->date('date');
+            $table->integer('start_minutes');
+            $table->integer('minutes');
+            $table->char('status_dict_id', 36)->collation('ascii_bin');
+
+            $table->foreign('facility_id')
+                ->references('id')
+                ->on('facilities')
+                ->restrictOnDelete();
+
+            $table->foreign('category_dict_id')
+                ->references('id')
+                ->on('positions')
+                ->restrictOnDelete();
+
+            $table->foreign('type_dict_id')
+                ->references('id')
+                ->on('positions')
+                ->restrictOnDelete();
+
+            $table->foreign('status_dict_id')
+                ->references('id')
+                ->on('positions')
+                ->restrictOnDelete();
+        });
+
+        Schema::create('meeting_attendants', function (Blueprint $table) {
+            $table->char('id', 36)->collation('ascii_bin')->primary();
+            $table->char('meeting_id', 36)->collation('ascii_bin');
+            $table->char('user_id', 36)->collation('ascii_bin');
+            $table->string('attendance_type', 36)->collation('ascii_bin');
+            $table->boolean('has_arrived')->nullable();
```

Comment:
Myślę że czas perfect nie jest tutaj uzasadniony, to brzmi jak pole aktualizowane natychmiast w momencie gdy ktoś przybywa. Może "present" albo "was_present", ewentualnie "arrived"? O, albo "attended", to jest to co się robi ze spotkaniami.

Reply:
jest konwencja nazewnictwa, gdzie pola bool się zaczynają od "is" lub "has". Z tego wychodzą trochę koślawe nazwy, ale zgodne z pewną konwencją

Reply:
No niefortunnie brzmi, to może is_attended, albo has_attended?

I w ogóle to tak, wiem że to jest zgodnie z zadaniem i z dokumentem który wcześnie reviewowałem nawet, ale najwyraźniej wcześniej mi nie przyszło do głowy że ta nazwa jest zła albo zapomniałem napisać. Nie wydaje mi się żeby to było nie do zmiany, no chyba że jest, albo że ktoś bardzo uważa że nie mam racji. Byłoby nie do zmiany na przykład gdyby te nazwy były już w DB, bo robić migrację tylko po to żeby zmienić nazwę pola na lekko lepiej brzmiącą to przesada. Dlatego piszę teraz a nie łaskawie dopiero jak jest tworzony model.

Reply:
a i tak możliwe, że to będzie słownik ostatecznie, typu presence/arrival_status

Reply:
has_attended wydaje się spoko, gdyby nie to, że już jesteśmy w tabeli meeting_attendants

czyli to jest trochę bardziej, czy uczestnik spotkania dotarł czy nie dotarł, określanie czy uczestnik uczestniczył jest trochę takie se też

Reply:
Ale jak już schodzimy na ten poziom to attendant to "a person who is present on a particular occasion" więc jak nie dotarł to nie jest attendant...
Kalendarz googlowy na przykład nazywa to "guest", nawet w kontekście służbowym. Ja myślę że attendant jest OK, ale has_attended też jest OK. Wprawdzie "attendant that has not attended" jest trochę dziwne, ale wiadomo o co chodzi.

Reply:
Zgadzam się z pierwszym instynktem @TPReal - was_present jest tu najczytelniejsze (w dodatku jest jasne, że przed spotkaniem takie pole nie może mieć wartości.

Co do konwencji is/has, to wg mnie z has chodzi stricte o aspekt posiadania, a nie o użycie jako czasownik posiłkowy. Podobnie jeśli mamy obiekt odzwierciedlający jakiś fakt z przeszłości, odmienienie is -> was jest wg. mnie całkowicie zgodne z konwencją.

Reply:
Inna rzecz że być może można oznaczyć że ktoś nie przyjdzie jeszcze zanim spotkanie się odbędzie, na przykład jeśli odwoła, wtedy czas przeszły jest średnio dobry. Ja bym się w sumie trzymał może is/has skoro taka konwencja obowiązuje.

Reply:
No to co Damian napisał to w sumie wszystko prawda, ale jednocześnie jakoś tak wydaje mi się, że to dobrze działa wizualnie jak są tylko te dwa "prefiksy". Można też uznać, że dodamy sobie tę lub podobną kolumnę jak będzie oznaczanie obecności na spotkaniach, i może jednak oparte na słowniku

Reply:
Jeśli ktoś odwoła to czy nie powinno się zawczasu usuwać z attendees po prostu? Jeśli nie to może faktycznie zmienić na cos w stylu "meeting_invitees" (albo invites) i już mieć pole `"attendence_status": "cancelled"|"present"|"no_show"`?

Reply:
pod ręką mam teamsy, i w nich to się nazywa uczestnicy, a dopiero potem ci uczestnicy mogą potwierdzić udział

no i jeżeli jestem umówiony na wizytę u lekarza, to jest jakby coś więcej niż to, że zostałem zaproszony

Reply:
Ah sorry, nie odświeżył mi się komentarz jak wysyłałem swój. To co, pomijam tę kolumnę i da nam to czas na przemyślenie jak chcemy to zapisywać?

Reply:
więc do nazwy tabeli jestem przekonany, do nazwy kolumny nie jestem, może po prostu status_dict_id i dodać słownik w tym momencie pusty, jeszcze ustalę jakie ma opcje (coś typu: obecny, spóźniony, odwołał, nie przyszedł)

Reply:
Wydaje się że słownik nas nie ominie. Bo oni nie robią tak że jak się spotkanie odwoła to je usuwają, tylko oznaczają jako odwołane, z powodem odwołania. A powód odwołania w szczególności może być że ktoś odwołał/nie przyszedł, i wtedy ta informacja również gdzieś powinna być utrwalona. Więc rzeczywiście tak jak Mikołaj mówi na razie.

Reply:
Zmieniłem na słownik